src/transformers/deepspeed.py

```diff
@@ -283,36 +160,28 @@ def trainer_config_finalize(self, args, model, num_training_steps):
             )
 
 
-# keep the config object global to be able to access it anywhere during TrainingArguments life-cycle
-_hf_deepspeed_config_weak_ref = None
-
-
-def set_hf_deepspeed_config(hf_deepspeed_config_obj):
-    # this is a special weakref global object to allow us to get to Deepspeed config from APIs
-    # that don't have an easy way to get to the Deepspeed config outside of the Trainer domain.
-    global _hf_deepspeed_config_weak_ref
-    # will go away automatically when HfDeepSpeedConfig is destroyed (when TrainingArguments is destroyed)
-    _hf_deepspeed_config_weak_ref = weakref.ref(hf_deepspeed_config_obj)
-
-
-def unset_hf_deepspeed_config():
-    # useful for unit tests to ensure the global state doesn't leak - call from `tearDown` method
-    global _hf_deepspeed_config_weak_ref
-    _hf_deepspeed_config_weak_ref = None
-
-
 def is_deepspeed_zero3_enabled():
-    if _hf_deepspeed_config_weak_ref is not None and _hf_deepspeed_config_weak_ref() is not None:
-        return _hf_deepspeed_config_weak_ref().is_zero3()
-    else:
-        return False
+    accelerator = Accelerator()
+    if hasattr(accelerator.state, "deepspeed_plugin"):
+        return (
+            accelerator.state.deepspeed_plugin.hf_ds_config.is_zero3()
+            and accelerator.state.deepspeed_plugin.zero3_init_flag
+        )
```

Comment:
The `Accelerator` object should not be created behind the user back as they might set their own options in it. We should instantiate `AcceleratorState` here (which is always the same) and use it directly.